### DISCUSSION TITLE
Reasoning behind putting a (cyclic-mode) DMA into a `busy=false` state after a TC interrupt was triggered in STM32 DMA driver?

### DISCUSSION THREAD

**Discussion by siwica** [zephyrproject-rtos/zephyr#65660] [Q&A]
We are working on generating digitals waveforms to control WS2812 LEDs and therefore need to fill the CCR register of a timer on a STM32 board via (memory -> peripheral, cyclic mode) DMA which in general works fine.

Unfortunately, we have encountered the problem that once the TC interrupt was triggered once,  the DMA is put into a mode (e.g. [here](https://github.com/zephyrproject-rtos/zephyr/blob/41b7c17ac4bd0198fc9bb3a0e55aa8d5e2fae96e/drivers/dma/dma_stm32.c#L114C3-L114C9) or [here](https://github.com/zephyrproject-rtos/zephyr/blob/41b7c17ac4bd0198fc9bb3a0e55aa8d5e2fae96e/drivers/dma/dma_stm32.c#L126C9-L126C9), depending on whether or not DMAMUX is used) (`stream->busy = false`) where it stops to call the specified callbacks on further {HT,TC}-interrupts, making it impossible for us to refill the DMA memory buffer after if was used up. This, however, is essential for our use case. 

What is the reasoning behind resetting the busy-state of the DMA's stream?

**Comment by erwango** [comment]
Looks you're right: https://github.com/zephyrproject-rtos/zephyr/pull/60909
